packages/block-library/src/cover/edit.js

```diff
@@ -187,33 +191,52 @@ class CoverEdit extends Component {
 						</>
 					) }
 				</BlockControls>
-				{ !! url && (
-					<InspectorControls>
-						<PanelBody title={ __( 'Cover Settings' ) }>
-							{ IMAGE_BACKGROUND_TYPE === backgroundType && (
+				<InspectorControls>
+					{ !! url && (
+						<PanelBody title={ __( 'Media Settings' ) }>
+							{ !! url && ( IMAGE_BACKGROUND_TYPE === backgroundType ) && (
 								<ToggleControl
 									label={ __( 'Fixed Background' ) }
 									checked={ hasParallax }
 									onChange={ toggleParallax }
 								/>
 							) }
-							{ IMAGE_BACKGROUND_TYPE === backgroundType && ! hasParallax && (
+							{ !! url && ( IMAGE_BACKGROUND_TYPE === backgroundType ) && ! hasParallax && (
 								<FocalPointPicker
 									label={ __( 'Focal Point Picker' ) }
 									url={ url }
 									value={ focalPoint }
 									onChange={ ( value ) => setAttributes( { focalPoint: value } ) }
 								/>
 							) }
-							<PanelColorSettings
-								title={ __( 'Overlay' ) }
-								initialOpen={ true }
-								colorSettings={ [ {
-									value: overlayColor.color,
-									onChange: setOverlayColor,
-									label: __( 'Overlay Color' ),
-								} ] }
-							>
+							{ !! ( url || id ) && (
```

Comment:
I guess we can remove the id check here, an image can not have an id without URL ( but can have a URL without an id).

Reply:
👍🏻